### ISSUE TITLE
Add obfs4

### ISSUE THREAD

**Issue by myleshorton** [getlantern/lantern#3568]
Particularly for pro users.

**Comment by uaalto** [comment]
I can look into this already. Since I'm soon at the Pluggable Transports workshop, it might be the right time. What do you think @myleshorton ?

**Comment by myleshorton** [comment]
Yes I think this will be a very cool fallback option for making sure pro users stay online. I *think* we probably want a separate set of servers that just implement obfs4 so as not to impact fingerprintability, but either way I think this would be a very powerful addition! I've been chatting with @oxtoacart about this as well.

**Comment by uaalto** [comment]
OK, so @oxtoacart you are working on this? I've been looking at obfs4 code, but haven't started integrating it yet.

**Comment by oxtoacart** [comment]
@uaalto I've not done any research or implementation on this, just chatted with @myleshorton.

**Comment by myleshorton** [comment]
To be clear we really shouldn't start work on this until after valencia is
released on desktop and mobile and until after pro itself is released!

On Tue, Feb 16, 2016 at 4:04 AM, oxtoacart <notifications@github.com> wrote:

> @uaalto <https://github.com/uaalto> I've not done any research or
> implementation on this, just chatted with @myleshorton
> <https://github.com/myleshorton>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/getlantern/lantern/issues/3568#issuecomment-184657159>
> .
>



-- 
--
President
Brave New Software Project, Inc.
https://www.getlantern.org
pgp A998 2B6E EF1C 373E 723F A813 045D A255 901A FD89

**Comment by uaalto** [comment]
ok!

**Comment by myleshorton** [comment]
Closing in favor of #3172
